vllm/model_executor/guided_decoding/__init__.py

```diff
@@ -34,8 +34,8 @@ def _adapt_request_for_tool_use(request: Union[CompletionRequest,
     if type(request) is CompletionRequest:
         return request
 
-    # user has chosen to not use any tool
-    if request.tool_choice == "none":
+    # user has chosen to not use any tool, OR is allowing the model to choose a tool.
+    if request.tool_choice == "none" or request.tool_choice == "auto":
```

Comment:
```python
request.tool_choice in ("none", "auto")
```

Reply:
holding off on cleaning up guided decoding for now because I have a couple of bugs to fix with it that are breaking tests, and it's highly likely my changes in one of these files is the source of it. I'm going to fix those first, then I'll clean them up